### DISCUSSION TITLE
When are translations updated in POEditor?

### DISCUSSION THREAD

**Discussion by FloEdelmann** [streetcomplete/StreetComplete#2518] [Repository Maintenance] [answered]
As noted in https://github.com/streetcomplete/StreetComplete/blob/master/CONTRIBUTING.md#translating-the-app:

> Before each release, translations are pulled in from POEditor.

But, when are new strings pulled from GitHub to POEditor?

**Comment by FloEdelmann** [comment]
POEditor's [GitHub integration docs](https://poeditor.com/help/how_to_translate_a_language_file_from_a_github_project) mentions that source strings can be automatically synchronized using web hooks: https://poeditor.com/help/how_to_use_the_github_webhook

Is that enabled?

**Reply by FloEdelmann** [reply]
@westnordost Would it be worth it to enable this automatic synchronization whenever new commits are pushed to the `master` branch?

**Comment by matkoniecz** [comment]
> But, when are new strings pulled from GitHub to POEditor?

From what I noticed it happens around beta release (on first beta release?).

> Is that enabled?

No.

**Comment by westnordost** [comment, marked as answer]
No, it's a gradle job

On 26 January 2021 09:16:54 CET, Flo Edelmann <notifications@github.com> wrote:
>@westnordost Would it be worth it to enable this automatic
>synchronization?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/streetcomplete/StreetComplete/discussions/2518#discussioncomment-310739

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Reply by FloEdelmann** [reply]
If I understand the code correctly, the gradle job is for pulling translations from POEditor to the local repo. My question was about the other direction: When are the strings in POEditor updated to reflect the changes in GitHub?

**Reply by westnordost** [reply]
Whenever I upload them. Usually I do this a day or so before the (beta) release.

**Reply by FloEdelmann** [reply]
Thanks for your answer!

I'll ask again since you seem to have misunderstood me earlier: Would it make sense to automate this import process with GitHub webhooks?
https://poeditor.com/help/how_to_use_the_github_webhook

**Reply by westnordost** [reply]
Two reasons come to my mind that speak against that:
1. sometimes strings are removed. If they are also immediately removed from POEditor, I need to be more cautious when exactly I merge to master features that remove strings, because then I lose the ability to update the translations in a minor update (that should not have that new feature) since those strings would then be missing
2. I **think** all contributors are notified if there are new strings. I want to keep the noise down. But I am not sure this is actually the case.

**Reply by matkoniecz** [reply]
> I think all contributors are notified if there are new strings.

Unfortunately they are not notified, at least I never was notified and even tried to find option for that some time ago.

**Reply by smichel17** [reply]
On Red Moon I use(d\*) [Weblate](https://weblate.org/); I have been very happy with it and would recommend it.

- Unlike POEditor, Weblate is itself [FLO](https://wiki.snowdrift.coop/about/free-libre-open#flo) software ([source code](https://github.com/WeblateOrg/weblate/)).

- I have it set up for continuous translation. One benefit of this is that translators are properly credited in the [commit log](https://github.com/LibreShift/red-moon/commits/master).

- It is a little unpolished (mostly: visually), but I have never observed or been told that that it impacts the usability.

- It has a nice translation badge/widget you can add to your README to show translation status: <a href="https://hosted.weblate.org/engage/red-moon/"><img src="https://hosted.weblate.org/widgets/red-moon/-/svg-badge.svg" alt="Translation status" /></a>
    <a href="https://hosted.weblate.org/engage/red-moon/?utm_source=widget">
    <img src="https://hosted.weblate.org/widgets/red-moon/-/horizontal-auto.svg" alt="Translation status" />
    </a>

\*Although Red Moon is [no longer under active development](https://github.com/LibreShift/red-moon/issues/281), it is still under active translation, with the last commit 4 days ago.

**Reply by westnordost** [reply]
Unless like almost everyone in the openstreetmap world would move to one single service, I'd not like to change the service because of this or that featur ethat might be missing or less optimal than on some other service.

**Reply by smichel17** [reply]
Fair enough, although to me switching from a proprietary tool to a FLO one is worth-it enough regardless of other factors.

It got lost in my final message, but originally I was just going to mention that if POEditor+webhooks gives individual commit attribution like Weblate does, then it may be worth enabling the feature for that, as long as it does not remove strings like you mentioned (but I would hope that it simply leaves the translations, unused, to be removed manually).

**Reply by mnalis** [reply]
> > I think all contributors are notified if there are new strings.
> Unfortunately they are not notified, at least I never was notified and even tried to find option for that some time ago.

Unfortunately I can confirm that, I never got any notifications on new texts for translation., which is a bummer. So unless I happen to check the translation page by accident, I'll only know that there is probably something to translate when github notifies me of new SC release, and by that time it it already too late to get translations in...

**Reply by smichel17** [reply]
![image](https://user-images.githubusercontent.com/6563664/106221383-6ed03800-61d5-11eb-86f0-0757c47dad99.png)
I found them — just had to go to my url bar and type `w` `e` `b` `l` `a` `t` `e` `.` `o` `r` `g` :stuck_out_tongue:

(I'm just joking around / giving you a hard time, don't mind me…)

**Comment by mnalis** [comment]
That looks like it could be beneficial, as it might help give translators more time for adding translations of new texts and then testing them in beta before the release goes out. (speaking with my translator hat on)

**Reply by westnordost** [reply]
Okay, then I guess automatically pushing hte strings from master to poeditor would make sense

**Reply by peternewman** [reply]
If you're worried about it deleting translations prematurely @westnordost , why not test it by adding a dummy string to translate to master (which could also be used to test the integration works), translate it in a few languages then remove it and check what happens?

As others have said, you'd hope there is an option to not remove them on import, either in POEditor or on the Gradle side, or only remove unused ones or something.
